File: tests-e2e/python/support/ui_auth.py
from playwright.sync_api import Page, expect

ADMIN_EMAIL = "admin@example.com"
ADMIN_PASSWORD = "admin123"


def login_as_admin(page: Page, base_url: str) -> None:
  page.goto(f"{base_url}/login")
  expect(page.get_by_test_id("login-form")).to_be_visible()
  page.get_by_test_id("login-email-input").fill(ADMIN_EMAIL)
  page.get_by_test_id("login-password-input").fill(ADMIN_PASSWORD)
  page.get_by_test_id("login-submit").click()
  expect(page.get_by_test_id("auth-email")).to_be_visible()
  expect(page.get_by_test_id("auth-email")).to_have_text(ADMIN_EMAIL)

File: tests-e2e/java/src/test/java/com/fettqa/events/e2e/support/UiAuth.java
package com.fettqa.events.e2e.support;

import static com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat;

import com.microsoft.playwright.Page;
import io.qameta.allure.Step;

public final class UiAuth {

  public static final String ADMIN_EMAIL = "admin@example.com";
  public static final String ADMIN_PASSWORD = "admin123";

  private UiAuth() {
  }

  @Step("Login as admin")
  public static void loginAsAdmin(Page page, String baseUrl) {
    login(page, baseUrl, ADMIN_EMAIL, ADMIN_PASSWORD);
  }

  @Step("Login with email: {email}")
  public static void login(Page page, String baseUrl, String email, String password) {
    page.navigate(baseUrl + "/login");
    assertThat(page.getByTestId("login-form")).isVisible();
    page.getByTestId("login-email-input").fill(email);
    page.getByTestId("login-password-input").fill(password);
    page.getByTestId("login-submit").click();
    assertThat(page.getByTestId("auth-email")).isVisible();
    assertThat(page.getByTestId("auth-email")).hasText(email);
  }
}
